frontend/src/pages/CompetitionDetails.js

```diff
@@ -242,16 +260,39 @@ const CompetitionDetails = () => {
     const fetchFolderContents = async () => {
       setIsLoading(true);
       try {
+        setLoading(true);
         const response = await fetch(item.url);
         const data = await response.json();
         setContents(data);
       } catch (error) {
         console.error('Error fetching folder contents:', error);
       } finally {
         setIsLoading(false);
+        setLoading(false);
       }
     };
 
+
+    if (loading) {
+      return (
+          <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', height: '400px' }}>
+            {/* Your existing loader */}
+            <ClipLoader color="#36D7B7" size={50} loading={loading} />
+
+            {/* Typewriter effect */}
+            <div style={{ marginTop: '20px', color: '#36D7B7', fontFamily: 'Courier New', fontSize: '20px' }}>
+              <Typewriter
+                  words={['Loading your data...', 'Fetching PRs...', 'Please wait...']}
+                  loop={true}
+                  typeSpeed={70}
+                  deleteSpeed={50}
+                  delaySpeed={1500}
+              />
+            </div>
+          </div>
+      );
+    }
```

Comment:
This is a more maintainable way 
`const LoadingScreen = () => {
    const loadingMessages = ['Loading your data...', 'Fetching PRs...', 'Please wait...'];
    
    return (
        <div 
            style={{
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'center',
                alignItems: 'center',
                height: '400px'
            }}
        >
            <ClipLoader 
                color="#36D7B7" 
                size={50} 
                loading={true} 
            />
            <div 
                style={{
                    marginTop: '20px',
                    color: '#36D7B7',
                    fontFamily: 'Courier New',
                    fontSize: '20px'
                }}
            >
                <Typewriter
                    words={loadingMessages}
                    loop={true}
                    typeSpeed={70}
                    deleteSpeed={50}
                    delaySpeed={1500}
                />
            </div>
        </div>
    );
};

// Usage
if (loading) {
    return <LoadingScreen />;
}`